portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -135,6 +159,45 @@ private int getNumComments(HttpServletRequest request) {
     return numComments;
   }
 
+  /** Return the URL of the uploaded image (null if no upload or if not image). 
+   * (Referenced FormHandlerServlet.java in hello-world-fetch.)
+   */
+  private String getBlobKey(HttpServletRequest request, String formElementID) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    // Blobstore maps form element ID to a list of keys of the blobs uploaded by the form.
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formElementID);
+
+    // User submitted form without file selected, so no blobKey.
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Form only allowed for one Blob input, so get first blobkey.
+    BlobKey blobKey = blobKeys.get(0);
+    
+    // User submitted form with no file, but an empty key was created--delete it.
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Check to make sure the file uploaded was an image.
+    // If the first five letters of the type are not "image," then return null and delete Blob from storage.
+    String fileType = blobInfo.getContentType();
```

Comment:
How is this content-type set? Does the blobstore client library figure this out from the file uploaded?

Reply:
Yeah, according to the documentation, this func returns the MIME Content type provided in the HTTP header during the file upload. https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/blobstore/BlobInfo